File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    profit=-Infinity,minn=prices[0];
    for(let i=1;i<prices.length;i++){
        profit=Math.max(profit,prices[i]-minn);
        minn=Math.min(minn,prices[i]);
    }
    return profit>0?profit:0;
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int res=Integer.MIN_VALUE;
        int min=prices[0];
        for(int i=1;i<prices.length;i++){
            res=Math.max(prices[i]-min,res);
            min=Math.min(prices[i],min);
        }
        return res>0?res:0;
    }
}